src/NServiceBus.Testing.Tests/HandlerTests.cs

```diff
@@ -461,110 +550,90 @@ public class Incoming : IMessage
 
         public class DeferringTimeSpanHandler : IHandleMessages<TestMessage>
         {
-            public IBus Bus { get; set; }
             public TimeSpan Defer { get; set; }
 
-            public void Handle(TestMessage message)
+            public async Task Handle(TestMessage message, IMessageHandlerContext context)
             {
-                Bus.Defer(Defer, message);
+                var sendOptions = new SendOptions();
+                sendOptions.DelayDeliveryWith(Defer);
+
+                await context.Send(message, sendOptions);
```

Comment:
Down below you have a few handler which only have on async operation in the handler and there you return the context task. Would it make sense to do the same here?